lua/astrocommunity/markdown-and-latex/markdown-preview-nvim/init.lua

```diff
@@ -1,7 +1,24 @@
 ---@type LazySpec
 return {
   "iamcco/markdown-preview.nvim",
-  build = function() vim.fn["mkdp#util#install"]() end,
+  build = function(plugin)
+    -- HACK: Use `yarn` when possible, or fallback to default install command
+    ---@see https://github.com/iamcco/markdown-preview.nvim/issues/690#issuecomment-2254280534
+    local package_manager = vim.fn.executable "yarn" and "yarn" or vim.fn.executable "npx" and "npx -y yarn" or false
+
+    if package_manager then
+      local cmd = string.format(
+        "!cd %s && cd app && COREPACK_ENABLE_AUTO_PIN=0 %s install --frozen-lockfile",
+        plugin.dir,
+        package_manager
+      )
+
+      vim.cmd(cmd)
+    else
+      vim.cmd [[Lazy load markdown-preview.nvim]]
+      vim.fn["mkdp#util#install"]()
```

Comment:
```suggestion
    local yarn = vim.fn.executable "yarn" == 1 and "yarn" or vim.fn.executable "npx" == 1 and "npx -y yarn" or false

    if yarn then
      vim.cmd(("!cd %s && cd app && COREPACK_ENABLE_AUTO_PIN=0 %s install --frozen-lockfile"):format(plugin.dir, yarn))
    else
      vim.notify("`yarn` or `npx` must be in the PATH to build `markdown-preview.nvim`", vim.log.levels.ERROR)
```

If we are checking for `yarn` and `npx` then we really don't have a reason to fall back on the install util at all. We can just throw an error to the user and move on

Reply:
Yeah, I was thinking exactly in that. I'll implement it.

Reply:
I was not able _(or I don't know how)_ to render the message in the screen as a notification if the `build` process fails. So, I decided to simply thrown an exception, this way the user can see the actual error.

<img width="1275" alt="image" src="https://github.com/user-attachments/assets/13e6ff43-fc34-400d-99fd-0325aaacd352">

Also, I included this conflicting behavior in the `README`.